libs/cluster/Server/ClusterConfig.cs

```diff
@@ -288,6 +288,26 @@ public List<int> GetLocalPrimarySlots()
             return slots;
         }
 
+        /// <summary>
+        /// Retrieve a list of slots served by this node.
+        /// </summary>
+        /// <returns>List of slots.</returns>
+        public List<int> GetLocalSlots()
+        {
+            var primaryId = this.LocalNodeId;
+            List<int> slots = [];
+
+            if (primaryId != null)
+            {
+                for (var i = 0; i < MAX_HASH_SLOT_VALUE; i++)
+                {
+                    if (slotMap[i].workerId > 0 && workers[slotMap[i].workerId].Nodeid.Equals(primaryId, StringComparison.OrdinalIgnoreCase))
```

Comment:
If you want the slots owned by the local node it suffices to do slotMap[i].workerId == 1.  Besides this, are you aware of   config.GetSlotList(uint workerId)? I think it does what you want for workerId = 1. Can you use that method instead of introducing a new one?

Reply:
Thanks for pointing out this one, I missed GetSlotList and wasn't aware of that workerId == 1 meaning local node relationship, I'll update.